tests/test_model_functions.py

```diff
@@ -62,6 +62,35 @@ def test_graph_for_dae(self):
     #######################################################################
     # this section contains tests for each get_XXX function on Model
 
+    def test_get_derived_quantities(self, basic_model):
+        """ Tests Model.get_state_symbols() works correctly. """
+
+        derived_quantities = basic_model.get_derived_quantities()
+        print(derived_quantities)
+        assert len(derived_quantities) == 0
+        derived_quantities = basic_model.get_derived_quantities(namespace_uri=shared.OXMETA)
+        print(derived_quantities)
+        assert len(derived_quantities) == 0
+
+    def test_get_derived_quantities2(self, aslanidi_model):
+        """ Tests Model.get_state_symbols() works correctly. """
+
+        derived_quantities = aslanidi_model.get_derived_quantities()
+        assert len(derived_quantities) == 78
+        derived_quantities = aslanidi_model.get_derived_quantities(namespace_uri=shared.OXMETA)
+        assert str(derived_quantities) == '[_membrane$i_Stim]'
+        derived_quantities = aslanidi_model.get_derived_quantities(namespace_uri='http://example.com#')
+        assert len(derived_quantities) == 0
+
+    def test_get_derived_quantities3(self):
+        """ Tests Model.get_state_symbols() works correctly. """
+        model = shared.load_model('matsuoka_model_2003')
```

Comment:
I'd rather not add another large real model just to test this! If we need to, we can add to the simple test models. But my comment above may make this unnecessary.

This test feels like it's checking how CG wants to use the method, rather than checking the method itself. Which is probably better tested in CG.

Reply:
yes of course I'll remove it once it works, i'm just asking why it's not working as expected????

Reply:
it's gone now